watch.go

```diff
@@ -0,0 +1,112 @@
+package rosedb
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	ActionPut    = "Put"
+	ActionDelete = "Delete"
+)
+
+// Event
+type Event struct {
+	Action  string
+	Key     []byte
+	Value   []byte
+	BatchId uint64
+}
+
+// NewEvent generate a key-value change event after batch submission.
+func NewEvent(action string, key, value []byte, batchId uint64) *Event {
+	return &Event{
+		Action:  action,
+		Key:     key,
+		Value:   value,
+		BatchId: batchId,
+	}
+}
+
+// Watcher temporarily stores event information,
+// as it is generated until it is synchronized to DB's watch.
+//
+// If the event is overflow, It will remove the oldest data,
+// even if event hasn't been read yet.
+type Watcher struct {
+	queue eventQueue
+	mu    sync.RWMutex
+}
+
+func NewWatcher(capacity uint64) *Watcher {
+	if capacity <= 0 {
+		capacity = 1000 // default capacity
+	}
+	return &Watcher{
+		queue: eventQueue{
+			Events:   make([]*Event, capacity),
+			Capacity: capacity,
+		},
+	}
+}
+
+func (w *Watcher) Insert(e *Event) {
```

Comment:
This function will not exposed to caller? if so, the first letter will be lowercase.
`insertEvent` may be better.